### DISCUSSION TITLE
Open3E Kommplett-Installationsskript

### DISCUSSION THREAD

**Discussion by Philip-Wiege** [open3e/open3e#152] [Ideas]
Hallo zusammen,

wäre es eigentlich möglich, ein Linux-Skript z.B. für einen Raspberry Pi zu schreiben, welches Open3E in einem Rutsch installiert? Ich habe heute mal meine Produktivinstanz auf die PIP Version hochgezogen und das war schon etwas Einrichtungsaufwand.

Grüße
Philip

**Comment by philippoo66** [comment]
hmm ich schätze schon. müsste man sich mal mit befassen. kommt ja auch noch das sudo nano autostartservice und so rein. 

Wo du es grad gemacht hast - da hast du es doch noch 'frisch' im Gedächtnis. willst du es nicht mal versuchen? ChatGPT hilft bestimmt gern dabei ;-)

Man braucht ja auch ein System, mit dem man es ausprobieren kann. Also eine SD Karte, die man immer wieder zurück auf Ursprungszustand flasht, weil ich denke, dass man schon mehr als einen Versuch braucht...

Wo siehst du da Stolpersteine oder 'Herausforderungen'?

**Comment by flecke-m** [comment]
Ich würde an der Stelle nochmal auf die Docker Container hinweisen, wie man Docker installiert findet man in unter 1 Minute und mit der Dokumentation die ich geschrieben habe, sollte es nicht allzu lange dauern das am laufen zu haben.

Das schwierigste sehe ich noch das Konfigurieren des CAN Bus Adapters im udev [https://github.com/open3e/open3e/blob/master/bestPractices/USB%20Adapter%20and%20udev.md](url)

Aber das sollte man ja immer machen (ob Script oder Docker).

**Comment by MyHomeMyData** [comment]
Es sind nur 4 Befehlszeilen notwendig:
```
sudo apt install python3-virtualenv
mkdir ~/open3e && cd ~/open3e
python3 -m venv .venv && source .venv/bin/activate
pip install git+https://github.com/open3e/open3e.git
```
Die Aktualisierung von open3e mit einer Zeile:
```
pip install --upgrade git+https://github.com/open3e/open3e.git
```
Lohnt sich dafür ein Skript? Wenn das Skript aber auch open3e als Service automatisch einrichten würde, wäre es eine große Hilfe. Stelle ich mir aber ziemlich knifflig vor.

**Reply by philippoo66** [reply]
ChatGPT:

Hier ist ein Bash-Skript, das die beschriebenen Schritte automatisiert und den Service für Open3E einrichtet, die Berechtigungen anpasst, den Daemon neu lädt und den Service aktiviert:

```
#!/bin/bash

# Variablen für Anpassung
SERVICE_FILE_PATH="/lib/systemd/system/open3e.service"
SERVICE_CONTENT="[Unit]
Description=Open3e Service Script
After=multi-user.target

[Service]
Type=simple
ExecStartPre=/bin/sleep 5
Restart=on-failure
User=pi
ExecStart=/bin/bash -c 'cd /home/pi/open3e && /home/pi/open3e/.venv/bin/open3e @args'

[Install]
WantedBy=multi-user.target"

# Überprüfen, ob Skript als Root ausgeführt wird
if [[ $EUID -ne 0 ]]; then
   echo "Bitte führe dieses Skript als Root oder mit sudo aus."
   exit 1
fi

# Service-Datei erstellen
echo "Erstelle die Service-Datei..."
echo "$SERVICE_CONTENT" > "$SERVICE_FILE_PATH"

# Berechtigungen anpassen
echo "Passe die Berechtigungen der Service-Datei an..."
chmod 644 "$SERVICE_FILE_PATH"

# Systemd Daemon neu laden
echo "Lade den Systemd-Daemon neu..."
systemctl daemon-reload

# Service aktivieren
echo "Aktiviere den Service..."
systemctl enable open3e.service

echo "Fertig! Der Service open3e.service wurde eingerichtet und aktiviert."
```

Verwendung

Kopiere das Skript in eine Datei, z. B. setup_open3e.sh.
Mache das Skript ausführbar:

    chmod +x setup_open3e.sh

Führe das Skript mit Root-Rechten aus:

    sudo ./setup_open3e.sh

Das Skript erstellt die open3e.service-Datei mit den angegebenen Inhalten, setzt die Berechtigungen, lädt den Systemd-Daemon neu und aktiviert den Service.

**Reply by philippoo66** [reply]
aber fehlt dann nicht auch noch die Einrichtng des Adapters?

aber das könnte ChatGPT uns bestimmt auch noch einbauen ;-)

**Reply by philippoo66** [reply]
in das Skript kann man ja auch die 4 Zeilen von oben noch reinbauen (und die noch fehlende Einrichtung des Adapters). Dann packt man das Skript zum Projekt und das Setup ist ein einziger Befehl, der das Skript ausführt (ggf. muss man vorher manuell das Skript ausführbar machen - also 2 Befehle...). :-)
